app/models/concerns/scopes_for_timestamps.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+# Scopes for collecting objects created (or updated) on, before, after or
+# between a given "%Y-%m-%d" string(s).
+# Include in a model to have them available:
+#
+# include ScopesForTimestamps
+#
+# Examples: Observation.created_between("2006-09-01", "2012-09-01")
+#           Name.updated_after("2016-12-01")
+
+module ScopesForTimestamps
+  extend ActiveSupport::Concern
```

Comment:
Invited Nathan. 

I think this is a pretty good use case for a concern, however. There's not much likelihood of making the code more confusing with this —that's the usual objection, at least.

Reply:
Thanks! I think what bothers him is inheritance/dependency.

Reply:
Given that all these classes already inherit from AbstractModel and it only depends on created_at and updated_at wouldn't it make sense to just add these there rather than creating another file that folks have to hunt down and explicitly include in the class definition?

Reply:
Now that's an idea. 🤭 
Let me check if any of the models that inherit from AbstractModel lack an updated_at column. I doubt it, but plenty of models do lack those columns, especially since we switched to `has_many_through` associations. If not I'll move the scopes.